src/Rule/Callback.php

```diff
@@ -19,12 +23,21 @@ public function __construct(callable $callback)
 
     protected function validateValue($value, DataSetInterface $dataSet = null): Result
     {
-        $result = new Result();
         $callback = $this->callback;
-        /**
-         * @var $callbackResult Result
-         */
         $callbackResult = $callback($value, $dataSet);
+
+        if (!$callbackResult instanceof Result) {
+            throw new CallbackRuleException(
+                sprintf(
+                    'Return value of callback must be an instance of %s, %s returned.',
```

Comment:
Would be nice to encapsulate the exception message to class if you did create a new exception class